### PR TITLE
[prometheus] kube-state-metrics v1.3.1

### PR DESCRIPTION
**Pull request by jhyelton**
**What this PR does / why we need it**: Updating the prometheus chart to use a new version of kube-state-metrics.

**Special notes for your reviewer**: https://github.com/kubernetes/kube-state-metrics#container-image mentions that quay.io is the preferred registry for kube-state-metrics, so I'm changing the chart to pull from that repo rather than k8s.gcr.io.  k8s.gcr.io does not have an image for v1.3.0.

### CODE REVIEW

File: stable/prometheus/values.yaml [hunk 1]
```diff
@@ -239,8 +239,8 @@ kubeStateMetrics:
   ## kube-state-metrics container image
   ##
   image:
-    repository: k8s.gcr.io/kube-state-metrics
-    tag: v1.2.0
+    repository: quay.io/coreos/kube-state-metrics
```
Comment (gianrubio) [inline stable/prometheus/values.yaml:268]: please update the Readme.md

**Review by gianrubio — CHANGES_REQUESTED** [review]
Should update readme.md

**Comment by jhyelton** [comment]
@mgoodness Is there anything else you need to review/test this PR?  Or is there someone else I could tag for review?

**Comment by monotek** [comment]
will this fix the broken kubernetes-service-endpoints / monitoring-prometheus-kube-state-metrics which fails with "context deadline exceeded" from chart version 6.2.0?

**Comment by monotek** [comment]
By the way...

v.1.3.1 is available on gcr.io too:

See: https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/kube-state-metrics?gcrImageListsize=50

**Comment by gianrubio** [comment]
@jhyelton I'd like to merge your PR, could you reopen it?

**Comment by monotek** [comment]
Why was this merged?
The images is updated on gcr.io as well.

**Comment by gianrubio** [comment]
there's a [note](https://github.com/kubernetes/kube-state-metrics#container-image) on the kube-state-metrics repo explaining why

> Note: The recommended docker registry for kube-state-metrics is quay.io. kube-state-metrics on gcr.io is only maintained on best effort as it requires external help from Google employees.

**Comment by monotek** [comment]
OK, thanks :-)

**Pull request merged**